src/terminal/input/terminalInput.cpp

```diff
@@ -372,135 +360,113 @@ typedef std::function<void(const std::wstring_view)> InputSender;
-                                           (ctrl ? 0x200 : 0) |
-                                           (alt ? 0x400 : 0);
-
-            // Make sure the VKEY is an _exact_ match, and that the modifier
-            // bits also match. This handles the hypothetical case we get a
-            // keyscan back that's ctrl+alt+some_random_VK, and some_random_VK
-            // has bits that are a superset of the bits set for question mark.
-            const auto wasQuestionMark = vkey == questionMarkVkey && WI_AreAllFlagsSet(keyScanFromEvent, questionMarkKeyScan);
-            const auto wasSlash = vkey == slashVkey && WI_AreAllFlagsSet(keyScanFromEvent, slashKeyScan);
-
-            // If the key pressed was exactly the ? key, then try to send the
-            // appropriate sequence for a modified '?'. Otherwise, check if this
-            // was a modified '/' keypress. These mappings don't need to be
-            // changed at all.
-            if ((ctrl && alt) && wasQuestionMark)
-            {
-                sender(CTRL_ALT_QUESTIONMARK_SEQUENCE);
-                success = true;
-            }
-            else if (ctrl && wasQuestionMark)
-            {
-                sender(CTRL_QUESTIONMARK_SEQUENCE);
-                success = true;
-            }
-            else if ((ctrl && alt) && wasSlash)
-            {
-                sender(CTRL_ALT_SLASH_SEQUENCE);
-                success = true;
-            }
-            else if (ctrl && wasSlash)
-            {
-                sender(CTRL_SLASH_SEQUENCE);
-                success = true;
-            }
+            return MakeOutput(CTRL_QUESTIONMARK_SEQUENCE);
+        }
+        else if ((ctrl && alt) && wasSlash)
+        {
+            return MakeOutput(CTRL_ALT_SLASH_SEQUENCE);
+        }
+        else if (ctrl && wasSlash)
+        {
+            return MakeOutput(CTRL_SLASH_SEQUENCE);
         }
     }
 
-    return success;
+    return MakeUnhandled();
 }
 
-// Routine Description:
-// - Searches the input array of mappings, and sends it to the input if a match was found.
-// Arguments:
-// - keyEvent - Key event to translate
-// - keyMapping - Array of key mappings to search
-// - sender - Function to use to dispatch translated event
-// Return Value:
-// - True if there was a match to a key translation, and we successfully sent it to the input
-static bool _translateDefaultMapping(const KeyEvent& keyEvent,
-                                     const std::span<const TermKeyMap> keyMapping,
-                                     InputSender sender)
+TerminalInput::OutputType TerminalInput::MakeUnhandled() noexcept
 {
-    const auto match = _searchKeyMapping(keyEvent, keyMapping);
-    if (match)
-    {
-        sender(match->sequence);
-    }
-    return match.has_value();
+    return {};
+}
+
+TerminalInput::OutputType TerminalInput::MakeOutput(const std::wstring_view& str)
+{
+    StringType out;
+    out.append(str);
+    return out;

```

Comment:
huh, can't you `return OutputType{ StringType{ str } };` to rely on native construction rather than `append`?

Reply:
Ah... with the `memory_buffer` I had to use `append()`. Good point.